blog/2024-12-10-pydantic.md

```diff
@@ -0,0 +1,20 @@
+---
+author: Joost Lekkerkerker
+authorURL: https://github.com/joostlek
+authorImageURL: https://avatars.githubusercontent.com/u/7083755?v=4
+title: "Pydantic bump to v2"
+---
+
+Pydantic is a popular library for data validation in Python.
+30 june 2023, Pydantic v2 was released.
+This release is not backwards compatible with Pydantic v1.
+
+Home Assistant Core 2024.12 ships with the Pydantic v1 library.
+Over the last year, the community has been working on making sure the libraries used by Home Assistant Core are supporting both v1 and v2 of Pydantic.
+By supporting both versions, we made our transition to Pydantic v2 as smooth as possible.
+
+From 2025.1, Home Assistant Core is expected to ship with Pydantic v2.
+Every custom component that uses Pydantic must be updated to support Pydantic v2 to keep working in Home Assistant Core 2025.1.
```

Comment:
```suggestion
Every custom component that uses Pydantic must be updated to support Pydantic v2 or use the v1 backward compatibility namespace to keep working in Home Assistant Core 2025.1.
```

Reply:
Oh I just added the note about the migration with the shims